src/main/java/cart/domain/CartItem.java

```diff
@@ -5,24 +5,62 @@
 import java.util.Objects;
 
 public class CartItem {
+
+    public static final int MINIMUM_QUANTITY = 1;
+
     private Long id;
     private int quantity;
     private final Product product;
     private final Member member;
 
     public CartItem(Member member, Product product) {
         this.quantity = 1;
+
+        validateMember(member);
         this.member = member;
+
+        validateProduct(product);
         this.product = product;
     }
 
     public CartItem(Long id, int quantity, Product product, Member member) {
+        validateId(id);
         this.id = id;
+
+        validateQuantity(quantity);
         this.quantity = quantity;
+
+        validateProduct(product);
         this.product = product;
+
+        validateMember(member);
         this.member = member;
```

Comment:
R : 모든 검증이 다 끝난 이후에 값을 대입하는거는 어떻게 생각해 ?
```suggestion
        validateId(id);
        validateQuantity(quantity);
        validateProduct(product);
        validateMember(member);
        this.id = id;
        this.quantity = quantity;
        this.product = product;
        this.member = member;
```

A : 검증 로직이 복잡한 경우가 아니라서 여러 메서드로 분리하지 않는게 한 눈에 보기 더 쉽다고 느껴지는데 이런거는 어때 ?

```suggestion
        validate(id, quantity, product, member);
        this.id = id;
        this.quantity = quantity;
        this.product = product;
        this.member = member;
```

Reply:
생성자가 두 개 있어서 필드 대입하기 전마다 검증하도록 했었는데
이 방식이 가독성이 더 좋은 것 같아서 변경해뒀어!